aten/src/ATen/native/quantized/cpu/conv_serialization.h

```diff
@@ -0,0 +1,292 @@
+
+  std::string version = "2";
+  std::vector<at::Tensor> non_optional;
+  std::vector<c10::optional<at::Tensor>> optional;
+
+  // create a packed int8_t tensor for conv params
+  std::vector<int8_t> params_vec;
+  params_vec.push_back(kSpatialDim);
+  auto stride = params->stride().vec();
+  params_vec.insert(params_vec.end(), stride.begin(), stride.end());
+  auto padding = params->padding().vec();
+  params_vec.insert(params_vec.end(), padding.begin(), padding.end());
+  auto dilation = params->dilation().vec();
+  params_vec.insert(params_vec.end(), dilation.begin(), dilation.end());
+  // output_padding is not implemented yet, so we fill in a default value
+  for (int i = 0; i < kSpatialDim; i++) {
+    params_vec.push_back(0);
+  }
+  params_vec.push_back(params->groups());
+  // transpose is not implemented yet, so we fill in a default value
+  params_vec.push_back(0);
+  int64_t vec_size = params_vec.size();
+  at::Tensor params_tensor = at::from_blob(
+      params_vec.data(), {vec_size},
+      at::TensorOptions().dtype(at::kChar))
+    // clone to retain ownership of the data
+    .clone();
+
+  at::Tensor weight;
+  c10::optional<at::Tensor> bias;
+  std::tie(weight, bias) = params->unpack();
+
+  non_optional.emplace_back(std::move(params_tensor));
+  non_optional.emplace_back(std::move(weight));
+  optional.emplace_back(std::move(bias));
+
+  return std::tie(version, non_optional, optional);
+}
+
+template <uint32_t kSpatialDim>
+ConvParamsSerializationTypeLegacy serialize_conv_legacy(
+    const c10::intrusive_ptr<ConvPackedParamsBase<kSpatialDim>>& params) {
+      at::Tensor weight;
+      c10::optional<at::Tensor> bias;
+      std::tie(weight, bias) = params->unpack();
+      torch::List<at::Tensor> stride;
+      torch::List<at::Tensor> padding;
+      torch::List<at::Tensor> dilation;
+      at::Tensor groups;
+      for (int64_t s : params->stride()) {
+        stride.emplace_back(at::tensor(s));
+      }
+      for (int64_t p : params->padding()) {
+        padding.emplace_back(at::tensor(p));
+      }
+      for (int64_t d : params->dilation()) {
+        dilation.emplace_back(at::tensor(d));
+      }
+      groups = at::tensor(params->groups());
+      return std::make_tuple(
+          std::move(weight),
+          std::move(bias),
+          stride,
+          padding,
+          dilation,
+          groups);
+}
+
+template <uint32_t kSpatialDim>
+c10::intrusive_ptr<ConvPackedParamsBase<kSpatialDim>> deserialize_conv(
+    ConvParamsSerializationType state) {
+
+  std::string version;
+  std::vector<at::Tensor> non_optional;
+  std::vector<c10::optional<at::Tensor>> optional;
+
+  std::tie(version, non_optional, optional) = state;
+
+  // c10::optional<at::Tensor> bias;
+  torch::List<int64_t> stride, padding, dilation;
+  int64_t groups;
+
+  at::Tensor conv_params_packed = non_optional[0];
+  at::Tensor weight = non_optional[1];
+  c10::optional<at::Tensor> bias = optional[0];
```

Comment:
You're assuming version 2 here.  I would add an assert for that.

Reply:
hmm, this is enforced by the input type to the function, `ConvParamsSerializationType` is for v2+, and `ConvParamsSerializationTypeLegacy` is for v1.  Since looks like it's not 100% intuitive, sg on adding the assert.